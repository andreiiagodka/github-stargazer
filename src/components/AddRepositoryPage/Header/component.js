import React from 'react'

import { Row, Col } from 'react-bootstrap'

const Header = () => (
  <Row>
    <Col md={2}></Col>
    <Col md={8} className='text-center'>
      <h5>{'Add Repository'}</h5>
    </Col>
    <Col md={2}></Col>
  </Row>
)

export default Header