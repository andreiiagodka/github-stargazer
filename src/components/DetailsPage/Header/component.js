import React from 'react'
import PropTypes from 'prop-types'

import { Row, Col, Button } from 'react-bootstrap'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faTrash } from '@fortawesome/free-solid-svg-icons'

import { deleteRepository } from '../../../shared/firebase'

const Header = ({ repository, history }) => {
  const handleDelete = () => {
    deleteRepository(repository.id)
    history.push('/')
  }

  return (
    <Row>
      <Col md={2}></Col>
      <Col md={8} className='text-center'>
        <h5>{repository.full_name}</h5>
      </Col>
      <Col md={2} className='text-right'>
        <Button variant='light' onClick={handleDelete}>
          <FontAwesomeIcon icon={faTrash} />
        </Button>
      </Col>
    </Row>
  )
}

Header.propTypes = {
  repository: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired
}

export default Header