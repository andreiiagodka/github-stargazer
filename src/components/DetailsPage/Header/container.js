import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { Row, Col, Button } from 'react-bootstrap'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faTrash } from '@fortawesome/free-solid-svg-icons'

import { connect } from 'react-redux'
import { withRouter } from 'react-router'
import * as actions from '../../../store/actions'

class Header extends Component {
  handleDelete = () => {
    const payload = {
      id: this.props.match.params.id,
      history: this.props.history
    }

    this.props.deleteRepository(payload)
  }

  render() {
    return (
      <Row>
        <Col md={2}></Col>
        <Col md={8} className='text-center'>
          <h5>{this.props.repository.full_name}</h5>
        </Col>
        <Col md={2} className='text-right'>
          <Button variant='light' onClick={this.handleDelete}>
            <FontAwesomeIcon icon={faTrash} />
          </Button>
        </Col>
      </Row>
    )
  }
}

Header.propTypes = {
  repository: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired
}

const mapDispatchToProps = dispatch => {
  return {
    deleteRepository: (id) => dispatch(actions.deleteRepository(id))
  }
}

export default connect(null, mapDispatchToProps)(withRouter(Header))