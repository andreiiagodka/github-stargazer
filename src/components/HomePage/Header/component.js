import React from 'react'

import { Row, Col, Button } from 'react-bootstrap'
import { Link } from 'react-router-dom'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faPlus } from '@fortawesome/free-solid-svg-icons'

const HeaderComponent = () => (
  <Row>
    <Col md={2}></Col>
    <Col md={8} className='text-center'>
      <h5>{'Github Stargazer'}</h5>
    </Col>
    <Col md={2} className='text-right'>
      <Link to='/new'>
        <Button variant='light'>
          <FontAwesomeIcon icon={faPlus} />
        </Button>
      </Link>
    </Col>
  </Row>
)

export default HeaderComponent
