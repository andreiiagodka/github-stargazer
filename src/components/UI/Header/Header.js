import React from 'react'
import PropTypes from 'prop-types'

import { Row, Col } from 'react-bootstrap'

const Header = ({ title, actionButton }) => (
  <Row>
    <Col md={2}></Col>
    <Col md={8} className='text-center'>
      <h5>{title}</h5>
    </Col>
    <Col md={2} className='text-right'>
      {actionButton}
    </Col>
  </Row>
)

Header.propTypes = {
  title: PropTypes.string.isRequired,
  actionButton: PropTypes.node
}

export default Header