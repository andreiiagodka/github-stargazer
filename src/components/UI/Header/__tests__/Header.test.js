import React from 'react'
import renderer from 'react-test-renderer'

import Header from '../Header'

describe('<Header /> component', () => {
  it('renders component correctly', () => {
    const props = {
      title: 'Title',
      actionButton: 'Action Button'
    }
    const tree = renderer.create(<Header {...props} />).toJSON()
  
    expect(tree).toMatchSnapshot()
  })
})
