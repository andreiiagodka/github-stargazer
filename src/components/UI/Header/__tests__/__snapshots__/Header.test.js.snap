// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> component renders component correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-md-2"
  />
  <div
    className="text-center col-md-8"
  >
    <h5>
      Title
    </h5>
  </div>
  <div
    className="text-right col-md-2"
  >
    Action Button
  </div>
</div>
`;
